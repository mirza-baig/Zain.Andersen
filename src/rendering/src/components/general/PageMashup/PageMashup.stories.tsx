// Global
import { Story, Meta } from '@storybook/react';
// Local
import PageMashup, { PageMashupProps } from './PageMashup';
import imagesForAllData from './PageMashup.mock-data-images-for-all';
import noImagesData from './PageMashup.mock-data-no-images';
import featuredImageOnlyData from './PageMashup.mock-data-featured-image';

export default {
  title: 'General/PageMashup',
  component: PageMashup,
} as Meta;
const Template: Story<PageMashupProps> = (props) => <PageMashup {...props} />;

export const ImagesForAll = Template.bind({});
ImagesForAll.args = imagesForAllData;

export const featuredImageOnly = Template.bind({});
featuredImageOnly.args = featuredImageOnlyData;

export const noImages = Template.bind({});
noImages.args = noImagesData;
