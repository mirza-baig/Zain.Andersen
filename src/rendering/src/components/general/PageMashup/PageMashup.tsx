// Global
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
// Components
import { Component } from 'src/helpers/Component';
import { getEnum } from 'lib/utils';
import { ComponentProps } from 'lib/types/component-props';
import Mashup from 'src/helpers/MashupHelpers/Mashup';

export type PageMashupProps =
  Feature.EnterpriseWeb.Enterprise.Components.General.PageMashup.PageMashup & ComponentProps;

const PageMashup = (props: PageMashupProps) => {
  return (
    <Component
      variant="full"
      padding="px-0"
      backgroundVariant={getEnum(props.fields?.backgroundColor)}
      dataComponent="general/pagemashup"
      {...props}
    >
      <Mashup {...props} />
    </Component>
  );
};

export default withDatasourceCheck()<PageMashupProps>(PageMashup);
