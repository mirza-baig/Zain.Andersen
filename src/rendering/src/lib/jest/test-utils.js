// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
