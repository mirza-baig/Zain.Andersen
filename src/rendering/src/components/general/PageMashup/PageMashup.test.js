// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import PageMashup from './PageMashup';
import imagesForAllData from './PageMashup.mock-data-images-for-all';

it('renders correctly', async () => {
  const component = snapshot(PageMashup, { componentProps: imagesForAllData });
  await hasDataComponent(component, 'general/pagemashup');
});
