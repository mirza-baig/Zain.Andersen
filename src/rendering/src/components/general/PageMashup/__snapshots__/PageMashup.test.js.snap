// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="general/pagemashup"
  id="id1751362294"
>
  <div
    class="my-8 w-full theme-gray bg-light-gray"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-0 gap-s md:gap-s "
    >
      <div
        class="col-span-12 py-l"
      >
        <div
          class="grid-rows-auto grid grid-cols-12 gap-s px-m md:max-w-screen-lg lg:mx-auto"
        >
          <div
            class="col-span-12 md:col-span-6"
          >
            <div
              class="items-top font-sans [&_a:hover]:underline text-theme-text text-s md:text-m font-heavy"
            >
              <h2>
                This is a featured headline lorem ipsum
              </h2>
            </div>
          </div>
          <div
            class="col-span-12 md:col-span-6"
          >
            <div
              class="text-theme-body text-body mb-m font-regular"
            >
              <div
                class="body-copy"
              >
                <div>
                  Body text goes here. Body text goes here. Body text goes here. Body text goes here. Body text goes here. Body text goes here. Body text goes here. 
                </div>
              </div>
            </div>
            <div
              class="mb-s mb-s flex items-start md:flex-row md:items-center"
            >
              <a
                aria-label="Home"
                class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed font-heavy"
                data-gtm-click=""
                data-gtm-dl-event="cta_click"
                href="http://"
                role="link"
                tabindex="0"
                target=""
              >
                Home
                <span
                  class="ml-xxs"
                />
              </a>
            </div>
          </div>
          <div
            class="col-span-12 grid grid-cols-12 gap-s md:gap-s"
          >
            <div
              class="order-last col-span-full flex items-center  md:order-none md:col-span-4 "
            >
              <div>
                <div
                  class="font-heavy text-theme-text text-xxs mb-xxs md:mb-s uppercase"
                >
                  <span>
                    Inspiration
                  </span>
                </div>
                <div
                  class="items-top font-sans [&_a:hover]:underline text-theme-text text-sm-m md:text-m font-heavy mb-s"
                >
                  <h3>
                    This is a card heading
                  </h3>
                </div>
                <div
                  class="body-copy text-theme-body text-body mb-m md:mb-l font-regular"
                >
                  <div>
                    This is a card description.
                  </div>
                </div>
                <div
                  class="md:mb-0 mb-s flex items-start md:flex-row md:items-center"
                >
                  <a
                    aria-label="Read More"
                    class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed font-heavy"
                    data-gtm-click=""
                    data-gtm-dl-event="cta_click"
                    href="/Article-Page-Mashup-1/"
                    role="link"
                    tabindex="0"
                    target=""
                    title=""
                  >
                    Read More
                    <span
                      class="ml-xxs"
                    />
                  </a>
                </div>
              </div>
            </div>
            <div
              class="col-span-full md:col-span-8"
            >
              <div
                class="relative hidden md:block aspect-picture   "
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 83.33333333333334% 0px 0px 0px;"
                  />
                  <img
                    alt=""
                    data-nimg="responsive"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <div
                class="relative block w-full md:hidden"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 83.33333333333334% 0px 0px 0px;"
                  />
                  <img
                    alt=""
                    data-nimg="responsive"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain; object-position: center;"
                  />
                  <noscript />
                </span>
              </div>
            </div>
          </div>
          <div
            class="relative col-span-12 flex gap-s border-b border-gray pt-xxs last:border-0 md:col-span-4 md:grid-cols-1 md:flex-col md:last:border-b"
          >
            <div
              class="relative hidden md:block   mb-xs"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27407%27%20height=%27270%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="a_awning-cat"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
            <div
              class="relative w-full md:hidden w-[155px] h-[155px] min-w-[155px] min-h-[155px] mb-xs"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27407%27%20height=%27270%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="a_awning-cat"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                />
                <noscript />
              </span>
            </div>
            <div
              class="flex-1 flex-col justify-between md:flex"
            >
              <div
                class="md:mb-s"
              >
                <div
                  class="font-heavy text-theme-text text-xxs mb-xxs uppercase"
                >
                  <span>
                    Generic eyebrow
                  </span>
                </div>
                <div
                  class="items-top font-sans [&_a:hover]:underline text-theme-text text-base font-serif font-heavy mb-xs"
                >
                  <h3>
                    This is a card headline
                  </h3>
                </div>
                <div
                  class="body-copy text-theme-body text-body mb-xxs md:mb-s font-regular line-clamp-3 md:line-clamp-none"
                >
                  <div>
                    This is a card description
                  </div>
                </div>
              </div>
              <div
                class="mb-s flex items-start md:flex-row md:items-center"
              >
                <a
                  aria-label="Read More"
                  class="hover:decoration-theme-btn-decoration flex w-fit items-center whitespace-normal font-sans text-text-link font-heavy text-theme-text hover:underline hover:underline-offset-8 disabled:border-gray disabled:text-gray"
                  data-gtm-click=""
                  data-gtm-dl-event="cta_click"
                  href="/Generic-Page-Mashup-1"
                  role="link"
                  tabindex="0"
                  target=""
                  title=""
                >
                  Read More
                  <span
                    class="ml-xxs"
                  />
                </a>
              </div>
            </div>
          </div>
          <div
            class="relative col-span-12 flex gap-s border-b border-gray pt-xxs last:border-0 md:col-span-4 md:grid-cols-1 md:flex-col md:last:border-b"
          >
            <div
              class="relative hidden md:block   mb-xs"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%271200%27%20height=%271000%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt=""
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
            <div
              class="relative block w-full md:hidden w-[155px] h-[155px] min-w-[155px] min-h-[155px] mb-xs"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27591%27%20height=%27591%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="591*591"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                />
                <noscript />
              </span>
            </div>
            <div
              class="flex-1 flex-col justify-between md:flex"
            >
              <div
                class="md:mb-s"
              >
                
                <div
                  class="items-top font-sans [&_a:hover]:underline text-theme-text text-base font-serif font-heavy mb-xs"
                >
                  <h3>
                    French Eclectic
                  </h3>
                </div>
                <div
                  class="body-copy text-theme-body text-body mb-xxs md:mb-s font-regular line-clamp-3 md:line-clamp-none"
                >
                  <div>
                    Description French Eclectic French Eclectic French Eclectic French Eclectic
                  </div>
                </div>
              </div>
              <div
                class="mb-s flex items-start md:flex-row md:items-center"
              >
                <a
                  aria-label="Read More"
                  class="hover:decoration-theme-btn-decoration flex w-fit items-center whitespace-normal font-sans text-text-link font-heavy text-theme-text hover:underline hover:underline-offset-8 disabled:border-gray disabled:text-gray"
                  data-gtm-click=""
                  data-gtm-dl-event="cta_click"
                  href="/Project-Showcase-Page/"
                  role="link"
                  tabindex="0"
                  target=""
                  title=""
                >
                  Read More
                  <span
                    class="ml-xxs"
                  />
                </a>
              </div>
            </div>
          </div>
          <div
            class="relative col-span-12 flex gap-s border-b border-gray pt-xxs last:border-0 md:col-span-4 md:grid-cols-1 md:flex-col md:last:border-b"
          >
            <div
              class="relative hidden md:block   mb-xs"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%271200%27%20height=%271000%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt=""
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
            <div
              class="relative block w-full md:hidden w-[155px] h-[155px] min-w-[155px] min-h-[155px] mb-xs"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27591%27%20height=%27591%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="591*591"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                />
                <noscript />
              </span>
            </div>
            <div
              class="flex-1 flex-col justify-between md:flex"
            >
              <div
                class="md:mb-s"
              >
                <div
                  class="font-heavy text-theme-text text-xxs mb-xxs uppercase"
                >
                  <span>
                    Inspiration
                  </span>
                </div>
                <div
                  class="items-top font-sans [&_a:hover]:underline text-theme-text text-base font-serif font-heavy mb-xs"
                >
                  <h3>
                    This is a card heading
                  </h3>
                </div>
                <div
                  class="body-copy text-theme-body text-body mb-xxs md:mb-s font-regular line-clamp-3 md:line-clamp-none"
                >
                  <div>
                    This is a card description.
                  </div>
                </div>
              </div>
              <div
                class="mb-s flex items-start md:flex-row md:items-center"
              >
                <a
                  aria-label="Read More"
                  class="hover:decoration-theme-btn-decoration flex w-fit items-center whitespace-normal font-sans text-text-link font-heavy text-theme-text hover:underline hover:underline-offset-8 disabled:border-gray disabled:text-gray"
                  data-gtm-click=""
                  data-gtm-dl-event="cta_click"
                  href="/Article-Page-Mashup-3"
                  role="link"
                  tabindex="0"
                  target=""
                  title=""
                >
                  Read More
                  <span
                    class="ml-xxs"
                  />
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
